kubernetes-ts-multicloud/aks.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as azure from "@pulumi/azure";
+import * as azuread from "@pulumi/azuread";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+import * as random from "@pulumi/random";
+import * as tls from "@pulumi/tls";
+
+export class AksCluster extends pulumi.ComponentResource {
+    public cluster: azure.containerservice.KubernetesCluster;
+    public provider: k8s.Provider;
+    public staticAppIP: pulumi.Output<string>;
+
+    constructor(name: string,
+                opts: pulumi.ComponentResourceOptions = {}) {
+        super("examples:kubernetes-ts-multicloud:AksCluster", name, {}, opts);
+
+        // Generate a strong password for the Service Principal.
+        const password = pulumi.secret(new random.RandomString("password", {
+            length: 20,
+            special: true,
+        }, {parent: this}).result);
```

Comment:
This is fine for now, but note that it is both not enough (the secret will still be in the checkpoint as the `id` and `result` of the resource creation), and too much (you shouldn't need this `pulumi.secret` call at all).

A more correct thing in the near term is to use:

```ts
{ parent: this, additionalSecretOutputs: ["result"] }
```

That will make the `result` field a secret (including ensuring it isn't serialized as part of the outputs).

Once we re-land https://github.com/pulumi/pulumi-terraform/issues/449, this won't be necessary.

Reply:
Fixed in https://github.com/pulumi/examples/pull/361